pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,619 @@
+        if isinstance(values, pa.Array):
+            self.data = pa.chunked_array([values])
+        elif isinstance(values, pa.ChunkedArray):
+            self.data = values
+        else:
+            raise ValueError(f"Unsupported type '{type(values)}' for ArrowStringArray")
+
+        if not pa.types.is_string(self.data.type):
+            raise ValueError(
+                "ArrowStringArray requires a PyArrow (chunked) array of string type"
+            )
+
+    @classmethod
+    def _chk_pyarrow_available(cls) -> None:
+        # TODO: maybe update import_optional_dependency to allow a minimum
+        # version to be specified rather than use the global minimum
+        if pa is None or LooseVersion(pa.__version__) < "1.0.0":
+            msg = "pyarrow>=1.0.0 is required for PyArrow backed StringArray."
+            raise ImportError(msg)
+
+    @classmethod
+    def _from_sequence(cls, scalars, dtype=None, copy=False):
+        cls._chk_pyarrow_available()
+        # convert non-na-likes to str, and nan-likes to ArrowStringDtype.na_value
+        scalars = lib.ensure_string_array(
+            scalars, na_value=cls._dtype.na_value, copy=False
+        )
+        return cls(pa.array(scalars, type=pa.string(), from_pandas=True))
+
+    @property
+    def dtype(self) -> ArrowStringDtype:
+        """
+        An instance of 'ArrowStringDtype'.
+        """
+        return self._dtype
+
+    def __array__(self, dtype=None) -> np.ndarray:
+        """Correctly construct numpy arrays when passed to `np.asarray()`."""
+        return self.to_numpy(dtype=dtype)
+
+    def __arrow_array__(self, type=None):
+        """Convert myself to a pyarrow Array or ChunkedArray."""
+        return self.data
+
+    def to_numpy(
+        self, dtype=None, copy: bool = False, na_value=lib.no_default
+    ) -> np.ndarray:
+        """
+        Convert to a NumPy ndarray.
+        """
+        # TODO: copy and na_value arguments are ignored
+        return self.data.__array__(dtype=dtype)
+
+    def __len__(self) -> int:
+        """
+        Length of this array.
+
+        Returns
+        -------
+        length : int
+        """
+        return len(self.data)
+
+    @classmethod
+    def _from_sequence_of_strings(cls, strings, dtype=None, copy=False):
+        return cls._from_sequence(strings, dtype=dtype, copy=copy)
+
+    @classmethod
+    def _from_factorized(cls, values, original):
+        return cls._from_sequence(values)
+
+    @classmethod
+    def _concat_same_type(cls, to_concat) -> ArrowStringArray:
+        """
+        Concatenate multiple ArrowStringArray.
+
+        Parameters
+        ----------
+        to_concat : sequence of ArrowStringArray
+
+        Returns
+        -------
+        ArrowStringArray
+        """
+        return cls(
+            pa.chunked_array(
+                [array for ea in to_concat for array in ea.data.iterchunks()]
+            )
+        )
+
+    def __getitem__(self, item: Any) -> Any:
```

Comment:
isn't this a pretty well defined output type? e.g. Union[str, NA,] or array of these?

Reply:
also related to #35259 (comment). If we don't consider subclasses and restrict dtype.na_value to be pd.NA, then this should be well defined.